### PR TITLE
remove auto set existing policy for HP VMs

### PR DESCRIPTION
**Pull request by liranr23**
This patch will remove the undesired behavior of setting auto pinning
policy to existing when the VM is set with High Performance profile and
not provided with any auto pinning policy input.

The existing policy is going to be blocked in ovirt, we should prevent
users from hitting an error while creating a VM without specifying a
value that might cause it.

### CODE REVIEW

File: ovirt/resource_ovirt_vm.go [hunk 1]
```diff
@@ -613,32 +613,23 @@ func (c *providerContext) resourceOvirtVMCreate(d *schema.ResourceData, meta int
...
 			"Please update to 4.4.5 or later.", engineVer.MustMajor(), engineVer.MustMinor(), engineVer.MustBuild())
-	} else {
-		// Mimic the UI behavior. unless specified set to existing policy for high performance VMs.
-		if _, ok := d.GetOk("auto_pinning_policy"); !ok && isHighPerformance {
-			err := d.Set("auto_pinning_policy", ovirtsdk4.AUTOPINNINGPOLICY_EXISTING)
+	} else if v, ok := d.GetOk("auto_pinning_policy"); ok {
+		autoPinningPolicy := ovirtsdk4.AutoPinningPolicy(fmt.Sprint(v))
+
+		// if we have a policy, we need to set the pinning to all the hosts in the cluster.
+		if autoPinningPolicy != ovirtsdk4.AUTOPINNINGPOLICY_DISABLED {
```
Comment (janoszen) [inline ovirt/resource_ovirt_vm.go:620]: This is in `resoruceOvirtVMCreate`. What about update?

File: ovirt/resource_ovirt_vm.go [hunk 2]
```diff
@@ -613,32 +613,23 @@ func (c *providerContext) resourceOvirtVMCreate(d *schema.ResourceData, meta int
 	if _, ok := d.GetOk("auto_pinning_policy"); ok && versionCompareResult < 0 {
 		log.Printf("[WARN] The engine version %d.%d.%d is not supporting the auto pinning feature. "+
 			"Please update to 4.4.5 or later.", engineVer.MustMajor(), engineVer.MustMinor(), engineVer.MustBuild())
-	} else {
-		// Mimic the UI behavior. unless specified set to existing policy for high performance VMs.
-		if _, ok := d.GetOk("auto_pinning_policy"); !ok && isHighPerformance {
-			err := d.Set("auto_pinning_policy", ovirtsdk4.AUTOPINNINGPOLICY_EXISTING)
+	} else if v, ok := d.GetOk("auto_pinning_policy"); ok {
```
Comment (janoszen) [inline ovirt/resource_ovirt_vm.go:616]: I think this if should be split into two, the second being the versionCompareResult. That would make the code much more readable.

File: ovirt/resource_ovirt_vm.go (hunk 1 above)
Comment (liranr23) [reply ovirt/resource_ovirt_vm.go:620]: we didn't support it in update. do you wish to add it?

File: ovirt/resource_ovirt_vm.go (hunk 2 above)
Comment (liranr23) [reply ovirt/resource_ovirt_vm.go]: I updated the code, let me know WDYT

**Review by janoszen — APPROVED** [review]
The tests that are fixed have passed, LGTM on my side.

File: ovirt/resource_ovirt_vm.go [hunk 3]
```diff
@@ -610,23 +610,15 @@ func (c *providerContext) resourceOvirtVMCreate(d *schema.ResourceData, meta int
...
-			if err != nil {
-				return err
-			}
-		}
-		if v, ok := d.GetOk("auto_pinning_policy"); ok {
-			isAutoPinning = true
-			autoPinningPolicy := ovirtsdk4.AutoPinningPolicy(fmt.Sprint(v))
-
+	if v, ok := d.GetOk("auto_pinning_policy"); ok {
+		autoPinningPolicy := ovirtsdk4.AutoPinningPolicy(fmt.Sprint(v))
```
Comment (Gal-Zaidman) [inline ovirt/resource_ovirt_vm.go:614]: Can you move this to the else section? or just use it in the condition?

**Comment by Gal-Zaidman** [comment]
LGTM from me :)

**Pull request merged**